File: src/lib/api/stackTemplate.ts
/**
 * 粘贴栈常用模板 API（P4）
 */
import { invoke } from "@tauri-apps/api/core";
import type { HistoryItem } from "@/stores/appStore";

export interface StackTemplateItem {
  itemType: string;
  text: string;
  content: string;
}

export interface StackTemplate {
  id: string;
  name: string;
  items: StackTemplateItem[];
  createdAt: string;
  usedAt: string | null;
}

/** 栈条目 → 模板条目快照（存独立拷贝，不是引用） */
export function stackItemsToTemplateItems(items: HistoryItem[]): StackTemplateItem[] {
  return items.map((it) => ({ itemType: it.type, text: it.text, content: it.content }));
}

export async function listStackTemplates(): Promise<StackTemplate[]> {
  return invoke<StackTemplate[]>("stack_template_list");
}

export async function saveStackTemplate(name: string, items: StackTemplateItem[]): Promise<string> {
  return invoke<string>("stack_template_save", { name, items });
}

export async function deleteStackTemplate(id: string): Promise<void> {
  await invoke("stack_template_delete", { id });
}

export async function touchStackTemplate(id: string): Promise<void> {
  await invoke("stack_template_touch", { id });
}

File: src-tauri/src/commands/stack_template.rs
//! 粘贴栈常用模板命令层（P4）。
//!
//! 仿 commands/chains.rs 的极简风格：list / save / delete 直接转发给 DataStore，
//! 验证都在 data_store 层做。

use crate::data_store::{DataStore, StackTemplate, StackTemplateItem};
use tauri::State;

#[tauri::command]
pub fn stack_template_list(store: State<DataStore>) -> Result<Vec<StackTemplate>, String> {
    store.stack_templates()
}

#[tauri::command]
pub fn stack_template_save(
    store: State<DataStore>,
    name: String,
    items: Vec<StackTemplateItem>,
) -> Result<String, String> {
    store.stack_template_save(&name, &items)
}

#[tauri::command]
pub fn stack_template_delete(store: State<DataStore>, id: String) -> Result<(), String> {
    store.stack_template_delete(&id)
}

#[tauri::command]
pub fn stack_template_touch(store: State<DataStore>, id: String) -> Result<(), String> {
    store.stack_template_touch(&id)
}
